### ISSUE TITLE
Generate hangs with fetch from Firebase to store

### ISSUE THREAD

**Issue by anishkny** [nuxt/nuxt.js#1058]
```
// pages/index.vue

// ...SNIP...

fetch({ store }) {
  return firebase.database().ref('/').once('value')
    .then((snap) => {
      return store.dispatch('setDataAction');
    });
}
```

```
// store/index.js

import Vuex from 'vuex'

const store = () => new Vuex.Store({
  state: {
    data: null,
  },
  mutations: {
    setDataMutation(state, _data) {
      state.data = _data;
    }
  },
  actions: {
    setDataAction({commit}, _data) {
      return new Promise((resolve, reject) => {
        commit('setDataMutation', _data);
        resolve();
      });
    },
  },
})

export default store
```

Hangs after executing `nuxt generate`:
```
...
  nuxt:generate Generate file: /index.html +906ms
  nuxt:generate HTML Files generated in 12.6s +1ms
[nuxt] Generate done      # <----- Hangs here
```

**Comment by Atinux** [comment]
Hi @anishkny 

It will be fixed in 1.0, see https://github.com/nuxt/nuxt.js/blob/dev/bin/nuxt-generate#L60

In the meantime: `Ctrl + C` or maybe closing the firebase connection at the end of `fetch`?

**Comment by anishkny** [comment]
Thanks for the reply @Atinux. Yup looks like when I patch nuxt-generate locally, it works! I will leave this issue open to communicate when the fix is released.

Btw, the Firebase connection is closed soon after data fetch because I am using `once()` which is effectively `on()` followed by `off()`. See [doc](https://firebase.google.com/docs/database/web/read-and-write#read_data_once).

Also, `Ctrl + C` does not work in CI but I am able to workaround.

For now, I am patching this change in `npm postinstall`. Shall wait for fix to be released!

```
--- node_modules/nuxt/bin/nuxt-generate	2017-07-09 11:04:44.000000000 -0400
+++ node_modules/nuxt/bin/nuxt-generate.patched	2017-07-09 11:05:47.000000000 -0400
@@ -24,6 +24,7 @@
   nuxt.generate()
     .then(() => {
       console.log('[nuxt] Generate done') // eslint-disable-line no-console
+      process.exit(0)
     })
     .catch((err) => {
       console.error(err) // eslint-disable-line no-console
```

PS: I am a huge fan of Nuxt which makes VueJS even more easier to use (didn't think such a thing was possible)!

**Comment by Atinux** [comment]
Close it since `rc3` is out
